src/syft/lib/pandas/frame.py

```diff
@@ -1,26 +1,34 @@
+"""Serde method for pd.DataFrame."""
+
 # third party
 import pandas as pd
+import pyarrow as pa
 
 # syft relative
 from ...generate_wrapper import GenerateWrapper
-from ...lib.python.dict import Dict
-from ...lib.python.primitive_factory import PrimitiveFactory
 from ...proto.lib.pandas.frame_pb2 import PandasDataFrame as PandasDataFrame_PB
 
 
 def object2proto(obj: pd.DataFrame) -> PandasDataFrame_PB:
-    pd_dict = PrimitiveFactory.generate_primitive(value=obj.to_dict())
-    dict_proto = pd_dict._object2proto()
+    """Convert pd.DataFrame to PandasDataFrame_PB with pyarrow.
+
+    # noqa: DAR101
+    # noqa: DAR201
 
-    return PandasDataFrame_PB(
-        id=dict_proto.id,
-        dataframe=dict_proto,
-    )
+    """
+    table = pa.Table.from_pandas(obj)
+    return PandasDataFrame_PB(dataframe=pa.serialize(table).to_buffer().to_pybytes())
 
 
 def proto2object(proto: PandasDataFrame_PB) -> pd.DataFrame:
-    dataframe_dict = Dict._proto2object(proto=proto.dataframe)
-    return pd.DataFrame.from_dict(dataframe_dict.upcast())
+    """Convert PandasDataFrame_PB to pd.DataFrame with pyarrow.
+
+    # noqa: DAR101
+    # noqa: DAR201
```

Comment:
NIPK: 

I believe documentation guys will not encourage this, what you can do instead is 

```
    """Proto to object conversion using to return desired model.

    Args:
        proto: Serialized version of Dataframe, which will be used to reconstruct model.

    Returns:
        Re-constructed dataframe.
    """
```

Reply:
Thansk @Param-29 !
That is the idea, not just ignoring "errors",  it is needed to addressed them

Reply:
Thanks for your review!